### PR TITLE
refactor: update example Dockerfile, publish script, and documentation

### PR DESCRIPTION
**Pull request by sjungling**
Notable changes
- Automatically downloads the latest version of the CLI from Maven central unless a Docker build arg is supplied. this should effectively deprecate the workflow in CLI https://github.com/moderneinc/moderne-cli/blob/main/.github/workflows/mass-ingest.yml
- Organizes all the relevant observability configuration into a single directory
- Changes the order of the layering so languages packs are next to last

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,31 +99,65 @@ If your organization does use Maven, you more than likely have shared configurat
...
+```bash
+docker build -t moderne-mass-ingest:latest \
+    --build-arg PUBLISH_URL=<> \
+    --build-arg PUBLISH_USER=<> \
+    --build-arg PUBLISH_PASSWORD=<> \
+    .
+```
+
+### Option 2: Connecting to the Moderne Platform
 ```bash
```
Comment (mike-solomon) [inline README.md:116]:
```suggestion
### Option 2: Connecting to the Moderne Platform

```bash
```

File: README.md [hunk 2]
```diff
@@ -93,31 +99,65 @@ If your organization does use Maven, you more than likely have shared configurat
...
     --build-arg PUBLISH_URL=<> \
     --build-arg PUBLISH_USER=<> \
     --build-arg PUBLISH_PASSWORD=<> \
+    --build-arg MODERNE_TENANT=<> \
+    --build-arg MODERNE_TOKEN=<> \
     .
 ```
 
+### Build Arguments
+| Argument | Description | Required |
```
Comment (mike-solomon) [inline README.md:127]:
```suggestion
### Build Arguments

| Argument | Description | Required |
```

File: repos.csv [hunk 3]
```diff
@@ -1,12 +1,2 @@
 cloneUrl,branch
-https://github.com/openrewrite/rewrite-cucumber-jvm,main
-https://github.com/openrewrite/rewrite-hibernate,main
 https://github.com/openrewrite/rewrite-javascript,main
```
Comment (mike-solomon) [inline repos.csv:2]: Why did we remove all of the repos here? I think it would be nice to include at least like 3 to show what that looks like.

Comment (sjungling) [reply repos.csv:2]: whoops. I was just removing to test. lemme add that back

**Pull request merged**